src/Model/Contact.php

```diff
@@ -3013,37 +3013,29 @@ public static function searchByName(string $search, string $mode = '', int $uid
 		}
 
 		// check supported networks
+		$networks = [Protocol::DFRN, Protocol::ACTIVITYPUB];
 		if (DI::config()->get('system', 'diaspora_enabled')) {
-			$diaspora = Protocol::DIASPORA;
-		} else {
-			$diaspora = Protocol::DFRN;
+			$networks[] = Protocol::DIASPORA;
 		}
 
 		if (!DI::config()->get('system', 'ostatus_disabled')) {
-			$ostatus = Protocol::OSTATUS;
-		} else {
-			$ostatus = Protocol::DFRN;
+			$networks[] = Protocol::OSTATUS;
 		}
 
+		$condition = ['network' => $networks, 'failed' => false, 'uid' => $uid];
+
 		// check if we search only communities or every contact
 		if ($mode === 'community') {
-			$extra_sql = sprintf(' AND `contact-type` = %d', self::TYPE_COMMUNITY);
-		} else {
-			$extra_sql = '';
+			$condition['contact-type'] = self::TYPE_COMMUNITY;
 		}
 
 		$search .= '%';
 
-		$results = DBA::p("SELECT * FROM `contact`
-			WHERE (NOT `unsearchable` OR `nurl` IN (SELECT `nurl` FROM `owner-view` where `publish` OR `net-publish`))
-				AND `network` IN (?, ?, ?, ?) AND
-				NOT `failed` AND `uid` = ? AND
-				(`addr` LIKE ? OR `name` LIKE ? OR `nick` LIKE ?) $extra_sql
-				ORDER BY `nurl` DESC LIMIT 1000",
-			Protocol::DFRN, Protocol::ACTIVITYPUB, $ostatus, $diaspora, $uid, $search, $search, $search
-		);
+		$condition = DBA::mergeConditions($condition,
+			["(NOT `unsearchable` OR `nurl` IN (SELECT `nurl` FROM `owner-view` where `publish` OR `net-publish`))
```

Comment:
Shouldn't we add a condition on the `blocked` and `deleted` fields?

Additionally, capitalization.
```suggestion
			["(NOT `unsearchable` OR `nurl` IN (SELECT `nurl` FROM `owner-view` WHERE `publish` OR `net-publish`))
```

Reply:
When we would exclude blocked contacts then we couldn't search for one to unblock it.

Reply:
I don't think that's a good reason to include them in the search, Blocked contacts appears in a specific place in the Contacts page, that's enough.

Reply:
Problem is: When we would exclude blocked accounts as well then we had to extend this function since we then had to consider not only the uid of the contacts but also the uid of the person viewing them. This would extend the PR too much I think.

Reply:
I meant contacts blocked at the node level.

Reply:
For this we had to modify the function as well since it can be called for public and personal contacts.

Reply:
it's the same field `blocked` in both cases, right? What would be the caveat to add the `blocked` field condition like you added the `deleted` condition?

Reply:
When a nonzero uid value is provided then we had to perform an additional query for the entry with the zero uid value.

Reply:
What if we don't do the additional query? Would it exclude expected matches with the condition on the `blocked` field? If not we don't need to perform the additional query (yet).